src/python/pants/core/goals/lint.py

```diff
@@ -195,27 +253,25 @@ async def lint(
                 f"{linters_with_multiple_reports}. The option `--lint-reports-dir` only works if "
                 f"each linter has a single result. {suggestion}"
             )
-        reports = itertools.chain.from_iterable(linter_to_reports.values())
         merged_reports = await Get(Digest, MergeDigests(report.digest for report in reports))
-        workspace.write_digest(merged_reports, path_prefix=lint_subsystem.reports_dir)
+        workspace.write_digest(merged_reports)
         logger.info(f"Wrote lint result files to {lint_subsystem.reports_dir}.")
 
     exit_code = 0
-    for result in sorted_results:
-        if result.exit_code == 0:
+    if all_results:
+        console.print_stderr("")
+    for results in all_results:
+        if results.skipped:
+            sigil = console.yellow("-")
+            status = "skipped"
+        elif results.exit_code == 0:
             sigil = console.green("✓")
             status = "succeeded"
         else:
             sigil = console.red("𐄂")
             status = "failed"
-            exit_code = result.exit_code
-        console.print_stderr(f"{sigil} {result.linter_name} {status}.")
-        if result.stdout:
-            console.print_stderr(result.stdout)
-        if result.stderr:
-            console.print_stderr(result.stderr)
-        if result != sorted_results[-1]:
-            console.print_stderr("")
+            exit_code = results.exit_code
```

Comment:
It may make sense at some point to include the error code for each errored execution, like so:
```
09:10:02.92 [WARN] Completed: Lint with Flake8 - failed (exit code 2).
```
This last one when many have failed hides alot of potentially useful information for tools that use expressive error codes.

Reply:
Done. Thanks for the suggestion!